server/index.js

```diff
@@ -41,6 +58,123 @@ app.post('/userEventData', (req, res) => {
+            id: curr.movieObj.id,
+            profile: curr.movieObj.profile,
+          },
+          progress: curr.progress,
+          startTime: new Date(curr.time - (6000000 * curr.progress)),
+        })
+        : prev;
+    }, []),
+  };
+};
+
+const buildRecommendationMsg = (msgObj) => {
+  return {
+    userId: msgObj.userId,
+    groupId: msgObj.groupId,
+    recs: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 && curr.movieObj.isRec === true && curr.progress === 1 ?
+        prev + curr.progress : prev;
+    }, 0),
+    nonRecs: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 && curr.movieObj.isRec === false && curr.progress === 1 ?
+        prev + curr.progress : prev;
+    }, 0),
+  };
+};
+
+const buildAppServerMsg = (msgObj) => {
+  return {
+    userId: msgObj.userId,
+    events: msgObj.events.reduce((prev, curr) => {
+      return curr.eventId === 3 ?
+        prev.concat({
+          movie: { id: curr.movieObj.id },
+          progress: curr.progress,
+          startTime: new Date(curr.time - (6000000 * curr.progress)),
+        })
+        : prev;
+    }, []),
+  };
+};
+
+let eventDashboard = [];
+app.post('/newEvent', (req, res) => {
+  db.addEvent(req.body.session, req.body.query)
+    .then((msgObj) => {
+      if (req.body.query.eventId === 4) {
+        sendMessages({
+          MessageBody: JSON.stringify(buildUserProfilesMsg(msgObj)),
+          QueueUrl: queues.userProfiles,
+          MessageGroupId: 'user-profile-events',
+        });
+      }
+      return msgObj;
+    })
+    .then((msgObj) => {
+      if (req.body.query.eventId === 4) {
+        sendMessages({
+          MessageBody: JSON.stringify(buildRecommendationMsg(msgObj)),
+          QueueUrl: queues.recommendation,
+          MessageGroupId: 'recommendation-events',
+        });
+      }
+      return msgObj;
+    })
+    .then((msgObj) => {
+      if (req.body.query.eventId === 4) {
+        sendMessages({
+          MessageBody: JSON.stringify(buildAppServerMsg(msgObj)),
+          QueueUrl: queues.appServer,
+          MessageGroupId: 'app-server-events',
+        });
+      }
+      return msgObj;
+    })
+    .then((data) => {
+      if (data) {
+        eventDashboard.push({
+          index: {
+            _index: 'user_events',
+            _type: 'event',
+          },
+        });
+        eventDashboard.push({
+          sessionId: req.body.session.id,
+          userId: req.body.session.userId,
+          groupId: req.body.session.groupId,
+          eventId: req.body.query.eventId,
+          progress: req.body.query.progress,
+          time: new Date(req.body.query.time),
+        });
+      }
+      if (req.body.query.eventId === 4) {
+        dashboard.elasticCreate(eventDashboard);
+        eventDashboard = [];
+      }
+      return data;
+    })
+    .then((data) => {
+      if (data) {
+        res.sendStatus(201);
+      }
+      return null;
+    })
+    .catch((err) => {
```

Comment:
This catch block is not capturing errors that may get generated by AWS SQS/dashboard functions.